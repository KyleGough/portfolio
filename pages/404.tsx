import { Divider } from '@components/Divider';
import { Layout } from '@components/Layout';
import { Link } from '@components/Link';
import { Section } from '@components/Section';
import React from 'react';

const NotFound: React.FC = () => (
  <Layout title="404 Page Not Found - Kyle Gough">
    <Section>
      <h1 className="mt-12 tracking-tight text-9xl font-extrabold text-center">
        404
      </h1>
      <h2 className="mb-8 tracking-tight text-6xl font-extrabold text-center">
        Page Not Found
      </h2>
    </Section>

    <Divider />

    <Section>
      <p className="text-center md:text-left mb-4">
        The requested page does not exist. You can use the following links to
        navigate:
      </p>
      <p className="text-center md:text-left md:ml-4">
        <Link to="/">Return to Homepage</Link>
      </p>
      <p className="text-center md:text-left md:ml-4">
        <Link to="/about">About Me</Link>
      </p>
      <p className="text-center md:text-left md:ml-4">
        <Link to="/projects">Projects</Link>
      </p>
      <p className="text-center md:text-left md:ml-4">
        <Link href="/CV.pdf">CV</Link>
      </p>
      <p className="text-center md:text-left md:ml-4">
        <Link href="https://github.com/KyleGough">GitHub</Link>
      </p>
    </Section>
  </Layout>
);

export default NotFound;
