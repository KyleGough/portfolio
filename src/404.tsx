import { useEffect } from 'react';
import { Link } from 'react-router-dom';
import Divider from './components/Divider';
import Section from './components/Section';

export default function NotFound() {

    useEffect(() => {
        document.title = '404 Page Not Found - Kyle Gough';
    }, [])

    return (
        <>
        <Section>
            <h1 className='mt-12 tracking-tight text-9xl font-extrabold text-center'>404</h1>
            <h2 className='mb-8 tracking-tight text-6xl font-extrabold text-center'>Page Not Found</h2>
        </Section>

        <Divider />

        <Section>            
            <p className='text-center md:text-left mb-4'>The requested page does not exist. You can use the following links to navigate:</p>
            <p className='text-center md:text-left md:ml-4'><Link to='/' className='text-link hover:text-link-hover'>Return to Homepage</Link></p>
            <p className='text-center md:text-left md:ml-4'><Link to='/about' className='text-link hover:text-link-hover'>About Me</Link></p>
            <p className='text-center md:text-left md:ml-4'><Link to='/projects' className='text-link hover:text-link-hover'>Projects</Link></p>
            <p className='text-center md:text-left md:ml-4'><a href='/CV.pdf' target='_blank' rel='noopener noreferrer' className='text-link hover:text-link-hover'>CV</a></p>
            <p className='text-center md:text-left md:ml-4'><a href='https://github.com/KyleGough' target='_blank' rel='noopener noreferrer' className='text-link hover:text-link-hover'>GitHub</a></p>
        </Section>
        </>
    );
}
