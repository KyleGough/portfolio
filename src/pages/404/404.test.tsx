import { render } from '@testing-library/react';
import React from 'react';
import { BrowserRouter } from 'react-router-dom';

import { NotFound } from './404';

describe('NotFound page', () => {
  it('renders', async () => {
    const { container } = render(
      <BrowserRouter>
        <NotFound />
      </BrowserRouter>
    );
    expect(container).toMatchSnapshot();
  });
});
