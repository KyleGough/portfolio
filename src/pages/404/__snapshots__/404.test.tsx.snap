// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound page renders 1`] = `
<div>
  <div
    class="first:pt-24 container text-primary py-8 last:pb-16"
  >
    <h1
      class="mt-12 tracking-tight text-9xl font-extrabold text-center"
    >
      404
    </h1>
    <h2
      class="mb-8 tracking-tight text-6xl font-extrabold text-center"
    >
      Page Not Found
    </h2>
  </div>
  <hr
    class="m-8 h-px bg-divider"
  />
  <div
    class="first:pt-24 container text-primary py-8 last:pb-16"
  >
    <p
      class="text-center md:text-left mb-4"
    >
      The requested page does not exist. You can use the following links to navigate:
    </p>
    <p
      class="text-center md:text-left md:ml-4"
    >
      <a
        class="text-link hover:text-link-hover focus:text-link-hover"
        href="/"
      >
        Return to Homepage
      </a>
    </p>
    <p
      class="text-center md:text-left md:ml-4"
    >
      <a
        class="text-link hover:text-link-hover focus:text-link-hover"
        href="/about"
      >
        About Me
      </a>
    </p>
    <p
      class="text-center md:text-left md:ml-4"
    >
      <a
        class="text-link hover:text-link-hover focus:text-link-hover"
        href="/projects"
      >
        Projects
      </a>
    </p>
    <p
      class="text-center md:text-left md:ml-4"
    >
      <a
        class="text-link hover:text-link-hover focus:text-link-hover"
        href="/CV.pdf"
        rel="noopener noreferrer"
        target="_blank"
      >
        CV
      </a>
    </p>
    <p
      class="text-center md:text-left md:ml-4"
    >
      <a
        class="text-link hover:text-link-hover focus:text-link-hover"
        href="https://github.com/KyleGough"
        rel="noopener noreferrer"
        target="_blank"
      >
        GitHub
      </a>
    </p>
  </div>
</div>
`;
